### PR TITLE
Merge fork - blueprinter-ruby/blueprinter to procore-oss/blueprinter

### PR DESCRIPTION
**Pull request by ritikesh**
Creating this PR to merge the community fork's changes to upstream. I've synced the repos and am creating this PR as-is for now, will cleanup and make changes based on review comments. Have consolidated the contributions into a single changelog entry as well.

Checklist:

* [x] I have updated the necessary documentation
* [x] I have updated the changelog, if necessary (CHANGELOG.md)
* [ ] I have signed off all my commits as required by [DCO](../CONTRIBUTING.md#legal)
* [x] My build is green

### CODE REVIEW

**Comment by jmeridth** [comment]
Gotta sign your commits now. Thank you so much for bringing this back into this repo. Looking forward to more contributions.

**Comment by ritikesh** [comment]
@jmeridth there are commits from multiple contributors since it's a forward merge as-is. If those aren't signed, not sure how we'd go about that. I don't want to squash them into a single commit and lose meaningful independent commits. What should we do about this then? Can there not be a one-time exception?

**Comment by jmeridth** [comment]
We can ignore DCO for that type of scenario. Thank you for clarifying.

File: .rubocop.yml
Comment (njbbaer) [inline .rubocop.yml:17]: 175 is quite long. How about 100?

Comment (ritikesh) [reply .rubocop.yml]: I tried both 100 and 115, but there are quite a few violations that might extend the scope of what is already a very large PR. Keeping it at 125 for now.

File: .rubocop.yml
Comment (jmeridth) [inline .rubocop.yml:2]: Do we want this to be 2.7 (the lowest we support) or 3.2 (the highest we support)?

Comment (lessthanjacob) [reply .rubocop.yml:2]: We'd want this to be the oldest version we support. From the [docs](https://docs.rubocop.org/rubocop/configuration.html#setting-the-target-ruby-version):

> Users may let RuboCop know the oldest version of Ruby which your project supports with:

```yaml
AllCops:
  TargetRubyVersion: 2.5
```

File: README.md
Comment (lessthanjacob) [inline README.md:987]:
```suggestion
gem 'blueprinter'
```

File: blueprinter.gemspec
Comment (lessthanjacob) [inline blueprinter.gemspec:32]: I see that we moved the location of gem dependencies from here to the `Gemfile` as part of [this PR](https://github.com/blueprinter-ruby/blueprinter/pull/10). Was that necessary to get everything working? Just trying to understand the intention of the change here!

File: .rubocop.yml
Comment (jmeridth) [reply .rubocop.yml:2]: We're not testing 2.5 anymore.  I'd suggest leaving to 2.7.  We should not support < 2.7.  We honestly shouldn't support 2.7, it's EOL.

**Comment by jmeridth** [comment]
We need to have https://github.com/procore-oss/blueprinter/pull/320 merge before this PR.  To ensure rubygem gets released.

File: blueprinter.gemspec
Comment (ritikesh) [reply blueprinter.gemspec:32]: development dependencies are better suited for Gemfiles right? it also helps when you want to use different versions of dev dependencies for different versions of ruby/rails you want to support. It keeps your gemspec clean and there's no material difference in terms of functionality.

Comment (lessthanjacob) [reply blueprinter.gemspec:32]: It does look like there's a pending default [Rubocop rule](https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecdevelopmentdependencies) for ensuring that development dependencies are enumerated in the `Gemfile` instead of the `gemspec`, so I'm happy to move forward with this change.

As @njbbaer mentioned though, `Appraisal` would be a really nice value add here for testing different Ruby versions. We can follow up on that, though.

**Review by lessthanjacob — APPROVED** [review]
🎉

**Comment by ritikesh** [comment]
@lessthanjacob can we merge this then?

**Comment by lessthanjacob** [comment]
@ritikesh Let's get one more +1, but after that, we should be good to merge!

**Review by jmeridth — APPROVED** [review]

**Comment by ritikesh** [comment]
thanks @lessthanjacob and @jmeridth. I'll post updates on the issue and the fork regarding this as well.

**Comment by ritikesh** [comment]
@lessthanjacob @jmeridth - looks like release workflow isn't working? I don't see the 0.3.0 version published to [Rubygems](https://rubygems.org/gems/blueprinter/) post merge

**Comment by lessthanjacob** [comment]
@ritikesh Releasing is a manual process currently (for example: https://github.com/procore-oss/blueprinter/pull/317).

I'll go ahead and open a PR to release `0.3.0` formally.

**Comment by ritikesh** [comment]
I see the release has happened but not sure if the workflow is fixed. Also, just wanted to know, how do we do communications within the maintainers?

**Comment by jmeridth** [comment]
@ritikesh I'll add you to the maintainers mailing list. 

Yes, release fixed and 0.30.0 is available on rubygems.org.

**Pull request merged**
